### PR TITLE
Add error handling and requirements.txt

### PR DESCRIPTION
**Pull request by mikaelforeseeti**

### CODE REVIEW

File: vanguard/client.py [hunk 1]
```diff
@@ -37,6 +37,7 @@
 class Client:
     def __init__(self, username, password, url, region="eu-central-1"):
         self.base_url = url
+        self.backend_url = "{}/backend"
```
Comment (maxwalls) [inline vanguard/client.py]: `f"{self.base_url}/backend"`

File: vanguard/client.py [hunk 2]
```diff
@@ -61,6 +62,8 @@ def authenticate(self, username, password, region, client_id, pool_id):
             region_name=region,
             config=Config(signature_version=botocore.UNSIGNED),
         )
+        if not client_id or not pool_id:
+            client_id, pool_id = self.cognito_params()
```
Comment (maxwalls) [inline vanguard/client.py]: We could just delete the `client_id` and `pool_id` parameters from this method and from `__init__` and only fetch them here

File: vanguard/client.py [hunk 3]
```diff
@@ -169,11 +172,12 @@ def wait_for_response(self, function, *args):
 
     def cognito_params(self):
         url = f"{self.base_url}/bundle.js"
-        pattern = r"{UserPoolId:\"(eu-central-1[^\"]+)\",ClientId:\"([^\"]+)\"}"
+        pattern = r"{\s*UserPoolId:\s*['\"](eu-central-1[^\"]+)['\"],\s*ClientId:\s*['\"]([^'\"]+)['\"]\s*}"
```
Comment (maxwalls) [inline vanguard/client.py]: `eu-central-1[^'\"]+`

File: vanguard/client.py (hunk 1 above)
Comment (mikaelforeseeti) [reply vanguard/client.py]: ...duh. of course. fixed.

File: vanguard/client.py (hunk 2 above)
Comment (mikaelforeseeti) [reply vanguard/client.py]: I agree, fixed.

File: vanguard/client.py (hunk 3 above)
Comment (mikaelforeseeti) [reply vanguard/client.py]: fixed

**Pull request merged**